tracing-subscriber/src/fmt/fmt_subscriber.rs

```diff
@@ -680,12 +687,16 @@ impl<C, N, E, W> Subscriber<C, N, E, W> {
 
 impl<C> Default for Subscriber<C> {
     fn default() -> Self {
+        // only enable ANSI when the feature is enabled, and the NO_COLOR
+        // environment variable is unset or empty.
+        let ansi = cfg!(feature = "ansi") && env::var("NO_COLOR").map_or(true, |v| v.is_empty());
```

Comment:
nit, TIOLI: this could be
```suggestion
        let ansi = cfg!(feature = "ansi") && env::var("NO_COLOR").map_or(true, String::is_empty);
```
although the current style is also fine.

Reply:
Using `String::is_empty` requires the addition as an `as_ref()` otherwise I get the following error.  I'm leaving it as is for now just because rustfmt breaks the line after `let ansi =` if I make the change.

```txt
error[E0631]: type mismatch in function arguments
   --> tracing-subscriber/src/fmt/fmt_subscriber.rs:695:80
    |
695 |         let ansi = cfg!(feature = "ansi") && env::var("NO_COLOR").map_or(true, String::is_empty);
    |                                                                   ------       ^^^^^^^^^^^^^^^^
    |                                                                   |            |
    |                                                                   |            expected due to this
    |                                                                   |            found signature defined here
    |                                                                   required by a bound introduced by this call
    |
    = note: expected function signature `fn(String) -> _`
               found function signature `for<'a> fn(&'a String) -> _`
note: required by a bound in `Result::<T, E>::map_or`
   --> /Users/dmlary/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/result.rs:784:25
    |
784 |     pub fn map_or<U, F: FnOnce(T) -> U>(self, default: U, f: F) -> U {
    |                         ^^^^^^^^^^^^^^ required by this bound in `Result::<T, E>::map_or`
```

Reply:
ah, never mind then!